components/layout_2020/dom_traversal.rs

```diff
@@ -42,12 +42,12 @@ pub(super) enum Contents {
 
     /// Content of a `::before` or `::after` pseudo-element that is being generated.
     /// <https://drafts.csswg.org/css2/generate.html#content>
-    OfPseudoElement(Vec<PseudoElementContentItem>),
+    OfPseudoElement(WhichPseudoElement, Vec<PseudoElementContentItem>),
 }
 
 pub(super) enum NonReplacedContents {
     OfElement,
-    OfPseudoElement(Vec<PseudoElementContentItem>),
+    OfPseudoElement(WhichPseudoElement, Vec<PseudoElementContentItem>),
```

Comment:
Why these new fields? There are at least some cases where they are redundant since a parameter `pseudo_element_type` is passed together with a `contents` parameter.

Reply:
Hah. These aren't needed at all. I think I just got a bit overzealous here. 😅 I've fixed this.